openedx/core/djangoapps/programs/tasks.py

```diff
@@ -499,71 +507,93 @@ def _retry_with_custom_exception(username, course_run_key, reason, countdown):
     LOGGER.info(f"Running task award_course_certificate for user {user}")
     try:
         course_key = CourseKey.from_string(course_run_key)
-        # Get the cert for the course key and username if it's both passing and available in professional/verified
```

Comment:
I re-factored this try/catch, because having 54 lines of code in a try/catch block which has its own nested exception handling is non-ideal architecture, and is going to make debugging more difficult. However, please make sure I didn't miss anything with the re-factor!